bundles/org.openhab.binding.webthing/src/main/java/org/openhab/binding/webthing/internal/link/TypeMapping.java

```diff
@@ -0,0 +1,144 @@
+     */
+    public static ItemType toItemType(Property propertyMetadata) {
+        String type = "string";
+        @Nullable
+        String tag = null;
+
+        switch (propertyMetadata.typeKeyword) {
+            case "AlarmProperty":
+            case "BooleanProperty":
+            case "LeakProperty":
+            case "LockedProperty":
+            case "MotionProperty":
+            case "OnOffProperty":
+            case "PushedProperty":
+                type = "switch";
+                tag = "Switchable";
+                break;
+            case "CurrentProperty":
+            case "FrequencyProperty":
+            case "InstantaneousPowerProperty":
+            case "VoltageProperty":
+                type = "number";
+                break;
+            case "HeatingCoolingProperty":
+            case "ImageProperty":
+            case "VideoProperty":
+                type = "string";
+                break;
+            case "BrightnessProperty":
+            case "HumidityProperty":
+                type = "dimmer";
+                break;
+            case "ColorModeProperty":
+                type = "string";
+                tag = "lighting";
+                break;
+            case "ColorProperty":
+                type = "color";
+                tag = "Lighting";
+                break;
+            case "ColorTemperatureProperty":
+                type = "dimmer";
+                tag = "Lighting";
+                break;
+            case "OpenProperty":
+                type = "contact";
+                tag = "ContactSensor";
+                break;
+            case "TargetTemperatureProperty":
+                type = "number";
+                tag = "TargetTemperature";
+                break;
+            case "TemperatureProperty":
+                type = "number";
+                tag = "CurrentTemperature";
+                break;
+            case "ThermostatModeProperty":
+                break;
+            case "LevelProperty":
+                if ((propertyMetadata.unit != null)
+                        && propertyMetadata.unit.toLowerCase(Locale.ENGLISH).equals("percent")) {
+                    type = "dimmer";
+                } else {
+                    type = "number";
+                }
+                break;
+            default:
+                switch (propertyMetadata.type.toLowerCase(Locale.ENGLISH)) {
+                    case "boolean":
+                        type = "switch";
+                        tag = "Switchable";
+                        break;
+                    case "integer":
+                    case "number":
+                        type = "number";
+                        break;
+                    default:
+                        type = "string";
+                        break;
+                }
+                break;
+        }
+
+        return new ItemType(type, tag);
+    }
+
+    /**
+     * The item type description
+     */
+    @NonNullByDefault
```

Comment:
Inner classes don't need to be annotated.

Reply:
removed